dali/operators/util/normal_distribution_op.h

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_UTIL_NORMAL_DISTRIBUTION_OP_H_
+#define DALI_OPERATORS_UTIL_NORMAL_DISTRIBUTION_OP_H_
+
+#include <random>
+#include <vector>
+#include "dali/core/convert.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+namespace detail {
+
+const std::string kMean = "mean";      // NOLINT
+const std::string kStddev = "stddev";  // NOLINT
+const std::string kShape = "shape";    // NOLINT
+
+const int kNumOutputs = 1;
+auto kShapeDefaultValue = std::vector<int>{-1};
+
+}  // namespace detail
+
+template<typename Backend>
+class NormalDistribution : public Operator<Backend> {
+ public:
+  ~NormalDistribution() override = default;
+
+  DISABLE_COPY_MOVE_ASSIGN(NormalDistribution);
+
+ protected:
+  explicit NormalDistribution(const OpSpec &spec) :
+          Operator<Backend>(spec),
+          shape_(spec.GetArgument<std::remove_const_t<decltype(this->shape_)>>(detail::kShape)),
```

Comment:
What happens when shape argument is not available?
=

Reply:
General idea is that the operator, which has both input and `shape` is incorrectly invoked (there's proper err msg about it). Details about it are presented in ops docs.

To specifically answer your question: the `shape_` will recieve default value (`vector<int>{-1}`).
Plus, there's this thing:
```
  bool IsShapeArgumentProvided(const OpSpec &spec) {
    return spec_.HasArgument(detail::kShape);
  }
```
which checks, if user provided `shape`.